crates/core/src/routing_info/mod.rs

```diff
@@ -0,0 +1,225 @@
+        }
+    }
+
+    /// Get a handle to the partition-to-node routing table.
+    pub fn partition_routing(&self) -> PartitionRouting {
+        PartitionRouting {
+            sender: self.sender.clone(),
+            partition_to_node_mappings: self.inner.clone(),
+        }
+    }
+
+    async fn run(mut self) -> anyhow::Result<()> {
+        debug!("Routing information refresher started");
+
+        let update_interval = Configuration::pinned()
+            .common
+            .metadata_update_interval
+            .into();
+        let mut update_interval = tokio::time::interval(update_interval);
+        update_interval.set_missed_tick_behavior(MissedTickBehavior::Delay);
+        let mut cancel = pin!(cancellation_watcher());
+
+        loop {
+            tokio::select! {
+                _ = &mut cancel => {
+                    info!("Routing information refresher stopped");
+                    break;
+                }
+                Some(cmd) = self.receiver.recv() => {
+                    match cmd {
+                        Command::SyncRoutingInformation => {
+                            self.spawn_sync_routing_information_task();
+                        }
+                    }
+                }
+                _ = update_interval.tick() => {
+                    debug!("Refreshing routing information...");
+                    self.spawn_sync_routing_information_task();
+                }
+            }
+        }
+        Ok(())
+    }
+
+    fn spawn_sync_routing_information_task(&mut self) {
+        if !self
+            .inflight_refresh_task
+            .as_ref()
+            .is_some_and(|t| !t.is_finished())
+        {
+            let partition_to_node_mappings = self.inner.clone();
+            let metadata_store_client = self.metadata_store_client.clone();
+
+            let task = task_center().spawn_unmanaged(
+                crate::TaskKind::Disposable,
+                "refresh-routing-information",
+                None,
+                {
+                    async move {
+                        sync_routing_information(partition_to_node_mappings, metadata_store_client)
+                            .await
+                    }
+                },
+            );
+            self.inflight_refresh_task = task.ok();
+        };
+    }
+}
+
+pub fn spawn_partition_routing_refresher(
+    tc: &TaskCenter,
+    partition_routing_refresher: PartitionRoutingRefresher,
+) -> Result<TaskId, ShutdownError> {
+    tc.spawn(
+        TaskKind::MetadataBackgroundSync,
+        "partition-routing-refresher",
+        None,
+        partition_routing_refresher.run(),
+    )
+}
+
+async fn sync_routing_information(
+    partition_to_node_mappings: Arc<ArcSwap<PartitionToNodesRoutingTable>>,
+    metadata_store_client: MetadataStoreClient,
+) -> anyhow::Result<()> {
+    let scheduling_plan: Option<SchedulingPlan> = metadata_store_client
+        .get(SCHEDULING_PLAN_KEY.clone())
+        .await?;
```

Comment:
Maybe we also log on debug if an error occurred here. That way things might become easier to debug if the routing does not update.